autoload/jetpack.vim

```diff
@@ -162,43 +201,69 @@ function jetpack#bundle()
   call delete(s:optdir, 'rf')
   let destdir = s:optdir .. '/_'
 
-  call s:createbuf()
+  " Merge plugins if possible.
+  let merged_count = 0
+  let merged_files = {}
   for i in range(len(bundle))
     let pkg = bundle[i]
-    call s:setbufline(1, printf('Merging Plugins (%d / %d)', i, len(s:pkgs)))
-    call s:setbufline(2, s:progressbar(1.0 * i / len(s:pkgs) * 100))
-    call s:setbufline(i+3, printf('Merging %s ...', pkg.name))
+    call s:setbufline(1, printf('Merging Plugins (%d / %d)', merged_count, len(s:pkgs)))
+    call s:setbufline(2, s:progressbar(1.0 * merged_count / len(s:pkgs) * 100))
     let srcdir = pkg.path .. '/' .. pkg.subdir
     let srcfiles = filter(s:files(srcdir), "!s:ignorable(substitute(v:val, srcdir, '', ''))")
     let destfiles = map(copy(srcfiles), "substitute(v:val, srcdir, destdir, '')")
-    let dupfiles = filter(copy(destfiles), "filereadable(v:val)")
+    let dupfiles = filter(copy(destfiles), "has_key(merged_files, v:val)")
     if g:jetpack#optimization == 1 && dupfiles != []
       call filter(bundle, "v:val != pkg")
       call add(unbundle, pkg)
       continue
     endif
     for i in range(0, len(srcfiles) - 1)
-      let srcfile = srcfiles[i]
-      let destfile = destfiles[i]
-      call mkdir(fnamemodify(destfile, ':p:h'), 'p')
-      call s:copy(srcfile, destfile)
+      call s:copy(srcfiles[i], destfiles[i])
+      let merged_files[destfiles[i]] = v:true
     endfor
-    call s:setbufline(i+3, printf('Merged %s ...', pkg.name))
+    call s:setbufline(merged_count+3, printf('Merged %s ...', pkg.name))
+    let merged_count += 1
   endfor
 
+  " Copy plugins.
   for i in range(len(unbundle))
     let pkg = unbundle[i]
-    call s:setbufline(1, printf('Copy Plugins (%d / %d)', i+len(bundle), len(s:pkgs)))
-    call s:setbufline(2, s:progressbar(1.0 * (i+len(bundle)) / len(s:pkgs) * 100))
-    call s:setbufline(i+len(bundle)+3, printf('Copying %s ...', pkg.name))
+    call s:setbufline(1, printf('Copy Plugins (%d / %d)', i+merged_count, len(s:pkgs)))
+    call s:setbufline(2, s:progressbar(1.0 * (i+merged_count) / len(s:pkgs) * 100))
     let srcdir = pkg.path .. '/' .. pkg.subdir
     let destdir = s:optdir .. '/' .. pkg.name
     for srcfile in s:files(srcdir)
       let destfile = substitute(srcfile, srcdir, destdir, '')
-      call mkdir(fnamemodify(destfile, ':p:h'), 'p')
-      call s:copy(srcfile, destfile)
+      call s:copy(srcfile, substitute(srcfile, srcdir, destdir, ''))
+    endfor
+    call s:setbufline(i+merged_count+3, printf('Copied %s ...', pkg.name))
+  endfor
+endfunction
+
+function jetpack#diplay()
```

Comment:
As `jetpack#display` will never be called out of the script, how about `s:display`?

Reply:
Yes. It can be changed because it is only referencing the convention such as jetpack#update etc.